### PR TITLE
Transform sequences act as the identity when empty.

### PR DESCRIPTION
**Pull request by bogovicj**
* also override isIdentity method
* add tests

see #24

### CODE REVIEW

File: src/main/java/net/imglib2/realtransform/AbstractRealTransformSequence.java [hunk 1]
```diff
@@ -102,6 +103,12 @@ public int numTargetDimensions()
 		return nTarget;
 	}
 
+	@Override
+	public boolean isIdentity()
+	{
+		return transforms.isEmpty();
```
Comment (imagejan) [inline src/main/java/net/imglib2/realtransform/AbstractRealTransformSequence.java]: This is too restrictive, isn't it? If this sequence actually contains an identity transform, or if it contains only an `InvertibleRealTransform` and its inverse, I would expect `isIdentity` to return `true`.

Maybe we could make use of `RealViewsSimplifyUtils.isIdentity()` here, as in the other transforms?

Or, if this is to be kept, I vote for adding a comment to the Javadoc of `RealTransform#isIdentity`, stating that if `false` it *could* still be identity.

Comment (bogovicj) [reply src/main/java/net/imglib2/realtransform/AbstractRealTransformSequence.java]: Yea, thanks for mentioning this. 

I'm also inclined to using `RealViewsSimplifyUtils.isIdentity()`

**Comment by bogovicj** [comment]
In response to @imagejan 's comments above, I put some work toward simplifying `(Invertible)RealTransformSequence`s when possible.
Here's the work:
https://github.com/bogovicj/imglib2-realtransform/commit/5e553f7b9eb6056d76bf7f8fc5ec1125461f0c3e

What I did
1) Check if any transform that is added is the identity (with `isIdentity`), and don't add it if it is
2) "unpack" any sequences that are added to a sequence.
3) Check if a transform that is added can be pre-concatenated to the transform preceding it, and if do, do it

(1) and (2) make sense to me.  

(3) may be taking things to far, but is nice for optimization, I think.
To make (3) thread-safe, it will be necessary to add copies of transforms rather than the originals (since they could be modified).

**Comment by bogovicj** [comment]
With @axtimwalde , decided against having the sequence itself do the simplifying.  Rather it will `isIdentity` will only return true if every element of the sequence `isIdentity`.
New PR incoming...

**Pull request closed without merging**